File: C_LR_remainders.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
    int t;
    cin >> t; // Read the number of test cases

    for (int testCase = 0; testCase < t; ++testCase) {
        int n, m;
        cin >> n >> m; // Read n and m

        vector<int> a(n);
        for (int i = 0; i < n; ++i) {
            cin >> a[i]; // Read the elements of the array
        }

        string s;
        cin >> s; // Read the string s

        // Solution
        long long prod = 1;
        for (int num : a) { // Calculate the product of elements in the array
            prod *= num;
        }

        vector<int> ans;
        int r = -1;
        int l = 0;
        for (char command : s) {
            ans.push_back(prod % m);
            if (command == 'R') {
                prod /= a[r];
                r -= 1;
            } else {
                prod /= a[l];
                l += 1;
            }
        }

        for (int num : ans) {
            cout << num << " ";
        }
        cout << endl;
    }

    return 0;
}

File: C_LR_remainders.py
t = int(input())  # Read the number of test cases

def large_product_remainder(numbers, divisor):
    result = 1
    for num in numbers:
        result = (result * num) % divisor
        # print(num, result)
    return result

for _ in range(t):
    n, m = map(int, input().split())  # Read n and m
    a = list(map(int, input().split()))  # Read the elements of the array
    s = input()  # Read the string s

    # Solution
    # prod = 1
    # for num in a: # O(n)
    #     prod *= num

    ans = []
    r = n - 1
    l = 0
    for command in s: # O(n)
        remainder = large_product_remainder(a[l:r+1], m)
        ans.append(str(remainder))
        if command == "R":
            r -= 1
        else:
            l += 1

    print(" ".join(ans))
    # for num in ans: # O(n)
    #     print(num, end=" ")
    # print()
